### PR TITLE
Modified tests to make them Windows compliant

### PR DESCRIPTION
**Pull request by marcusnaslund**
Minor changes to tests to avoid roundoff errors and different thread scheduling that occurs in Windows.

### CODE REVIEW

**Comment by marcusnaslund** [comment]
`expected  equal to  '10.00000000', was '10.00000000'  [tolerance: 0.00001000]`...

File: test/math/FloatMatrixTest.ooc [hunk 1]
```diff
@@ -74,7 +75,13 @@ FloatMatrixTest: class extends Fixture {
 
 		this add("multiplication (scalar)", func {
 			matrix = createMatrix(3, 3, [1.0f, 0, 0, 0, 2.0f, 0, 0, 0, 3.0f])
-			checkAllElements(2.0f * matrix, [2.0f, 0, 0, 0, 4.0f, 0, 0, 0, 6.0f])
+			product := 2.0f * matrix
+			expect(product[0, 0] > 2.0f - this tolerance, is true) //FIXME checkAllElements fails here for some reason
+			expect(product[0, 0] < 2.0f + this tolerance, is true)
```
Comment (davidhesselbom) [inline test/math/FloatMatrixTest.ooc]: What about `expect(product[0, 0] abs()` something, at the very least?

File: test/math/FloatMatrixTest.ooc [hunk 2]
```diff
@@ -145,7 +152,7 @@ FloatMatrixTest: class extends Fixture {
 		// 2 5
 		for (x in 0 .. matrix dimensions width) {
 			for (y in 0 .. matrix dimensions height) {
-				expect(matrix[x, y], is equal to(values[x * matrix dimensions height + y]))
+				expect(matrix[x, y], is equal to(values[x * matrix dimensions height + y]) within(this tolerance))
```
Comment (davidhesselbom) [inline test/math/FloatMatrixTest.ooc]: Try storing the result of `matrix[x, y]` in a variable and using that as the first argument to `expect` and see if that works better:
```ooc
(element, value) := matrix[x, y]
expect(element, is equal to(value) within(this precision)
```

Comment (marcusnaslund) [inline test/math/FloatMatrixTest.ooc]: Gah, now the original thing works on both platforms. I think the original problem was `this tolerance` was accidentally a `Double` which is problematic sometimes, but not always. Hang on.

**Comment by marcusnaslund** [comment]
Updated. As an extra bonus, I got rid of four curly brackets.

File: test/math/FloatMatrixTest.ooc [hunk 3]
```diff
@@ -7,6 +7,7 @@ FloatMatrixTest: class extends Fixture {
 	matrix := FloatMatrix new (3, 3)
 	nonSquareMatrix := FloatMatrix new (IntSize2D new(2, 3))
 	nullMatrix := FloatMatrix new(0, 0)
+	tolerance := 0.00001f
```
Comment (davidhesselbom) [inline test/math/FloatMatrixTest.ooc]: - In just about every other test, this variable is called `precision`.
- Use scientific notation: `1.0e-5f´ (makes it easier to count the zeroes).

**Comment by marcusnaslund** [comment]
Done.

**Comment by davidhesselbom** [comment]
This should be the last thing we need to get things to work, right? I'll let the tests run before I merge, then.

**Comment by marcusnaslund** [comment]
I hope so. It might a be a good idea to cancel all AppVeyor builds before this one, otherwise we will wait a long time.

**Comment by davidhesselbom** [comment]
Let me guess - I'm the only one who can do that, right?

**Comment by marcusnaslund** [comment]
Yep, I see no cancel button like I do for my own branch.

**Comment by davidhesselbom** [comment]
Done. I must have cancelled at least 20 builds...

**Comment by davidhesselbom** [comment]
Unbelievable. AppVeyor succeeds, but Circle fails. Is is Monday already?

**Comment by marcusnaslund** [comment]
Hmmmm.

**Comment by davidhesselbom** [comment]
`ThreadPool` now consistently fails on Circle CI for some reason. It has nothing to do with this PR, so what's causing it?

**Comment by marcusnaslund** [comment]
It was green a second ago, now it says failed again. I also have trouble seeing updates from certain runs, and some seem to fail between tests. Is CircleCI having problems?

**Comment by davidhesselbom** [comment]
Isn't it great when a system you rely on to find your bugs has bugs? ;)

**Pull request merged**